### PR TITLE
21620-Modularize-Pharo-Launcher-dependent-Baselines

### PR DESCRIPTION
**Pull request by guillep**
Modularize Pharo Launcher dependent Baselines

- Add BaselineOfVersionner
- Add BaselineOfSpec
- Add BaselineOfPharoMinimal

- Fixed existing baselines to point to new ones

### CODE REVIEW

File: src/BaselineOfVersionner/BaselineOfVersionner.class.st [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	spec for: #common do: [
+		
+		spec baseline: 'SUnit' with: [spec repository: repository]..
+		
+		spec
+			package: 'Versionner-Core-Model';
+			package: 'Versionner-Core-Commands';
+			package: 'Versionner-Core-Announcements';
+			package: 'Versionner-Core-DependenciesModel';
+			package: 'Versionner-Spec-Browser';
```
Comment (jecisc) [inline src/BaselineOfVersionner/BaselineOfVersionner.class.st:23]: Shouldn't this one depend on Spec baseline created in the same PR?

File: src/BaselineOfSpec/BaselineOfSpec.class.st [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+BaselineOfSpec >> baseline: spec [ 
+    
+	<baseline>
+
+	| repository |
+	 
+	repository := self packageRepositoryURL.
+       
+	spec for: #common do: [
+		spec postLoadDoIt: #'postload:package:'.
```
Comment (jecisc) [inline src/BaselineOfSpec/BaselineOfSpec.class.st]: I do not see BaselineOfSpec>>postLoad:package: is that normal?

File: src/BaselineOfVersionner/BaselineOfVersionner.class.st (hunk 1 above)
Comment (guillep) [reply src/BaselineOfVersionner/BaselineOfVersionner.class.st:23]: Probably. I just extracted the baselines as they were. They had no dependency information before either.

File: src/BaselineOfSpec/BaselineOfSpec.class.st (hunk 2 above)
Comment (guillep) [reply src/BaselineOfSpec/BaselineOfSpec.class.st]: Thanks :) I was in rush mode.

**Comment by jecisc** [comment]
Error is:

```
	Loaded -> Keymapping-Tools-Spec-tonel.1 --- tonel:///builds/workspace/and_branch_Pipeline_PR-1130-B65QC6P3DMQGZKTF5F6BQVMWQYW4URFOZ5Y66FYIX46EVYBIEEHA/32/bootstrap/scripts/../../src --- cache
	Loaded -> WebBrowser-Core-tonel.1 --- tonel:///builds/workspace/and_branch_Pipeline_PR-1130-B65QC6P3DMQGZKTF5F6BQVMWQYW4URFOZ5Y66FYIX46EVYBIEEHA/32/bootstrap/scripts/../../src --- cache
	Loaded -> HelpSystem-Core-tonel.1 --- tonel:///builds/workspace/and_branch_Pipeline_PR-1130-B65QC6P3DMQGZKTF5F6BQVMWQYW4URFOZ5Y66FYIX46EVYBIEEHA/32/bootstrap/scripts/../../src --- cache
	Loaded -> Rubric-tonel.1 --- tonel:///builds/workspace/and_branch_Pipeline_PR-1130-B65QC6P3DMQGZKTF5F6BQVMWQYW4URFOZ5Y66FYIX46EVYBIEEHA/32/bootstrap/scripts/../../src --- cache
Finished atomic loadMessageNotUnderstood: receiver of "nextOrNil" is nil
UndefinedObject(Object)>>doesNotUnderstand: #nextOrNil
InputEventSensor>>nextEvent
HandMorph>>processEventsFromQueue:
HandMorph>>processEvents
```

I don't know if it is releated.

**Comment by guillep** [comment]
So just to summarize the problems in the original commits:
 - an atomic Metacello load will only load atomically the packages of the baseline and not its dependencies
 - a metacello baseline that is a dependency of many baselines will execute its pre/post loads so many times
 - of course that created a couple of problems because postloads were not designed to be executed more than once.

Now
 - a new baseline (Baseline Of Pharo) describes everything that should be added to the Pharo full build
 - baseline of IDE is atomic
 - Added guards in baseline of IDE to avoid executing the postload multiple times.

**Review by pavel-krivanek — APPROVED** [review]
approved (as soon as the tests are green)

**Pull request merged**